hw/ip/spi_device/dv/env/spi_device_scoreboard.sv

```
@@ -102,9 +102,7 @@ class spi_device_scoreboard extends cip_base_scoreboard #(.CFG_T (spi_device_env
                                     mem_addr - tx_base, item.a_data), UVM_MEDIUM)
         end
       end else if (mem_addr inside {[rx_base : rx_base + rx_limit]}) begin // RX address
-        if (write && channel == AddrChannel) begin
-          `uvm_error(`gfn, "unexpected write on RX mem")
-        end else if (!write && channel == DataChannel) begin
+        if (!write && channel == DataChannel) begin
```

these tx/rx_mem may only work for FW mode.

After I implemented this, we added a generic mem as follows. Maybe you can consolidate them or let FW mode uses tx/rx_mem and the other modes use exp_mem
https://github.com/lowRISC/opentitan/blob/2023653b220d6fd4dde5fd9055a6ee5eb0802f2b/hw/dv/sv/cip_lib/cip_base_scoreboard.sv#L22

---

I will add this in separate PR

---

This section doesn't support your usage. the write to this location will be silently dropped

if you want to get rid of this error for now and address it later, please add TODO and comment the error.

```
// TODO, ...
// `uvm_error(...)
```